### PR TITLE
Support creating survey response and answers when POSTing survey_visit

### PR DESCRIPTION
**Pull request by mikelynch**
When we POST the survey_visit, we want to be able to support also creating the survey response and answers in the same request.

### CODE REVIEW

**Comment by mikelynch** [comment]
@mzagaja , I think this does most of what's needed for this one. There are a couple of minor Rubocop complaints – not totally sure the best remedies for them (one is for a line which is too long, but I don't see how to make it shorter with the doubly-nested attributes; and one is on a `has_many` relationship, where I'm not sure of the appropriate `:dependent` option).

One remaining question: should we be trying to validate the survey_answers? I.e. if the corresponding survey_question is multiple choice, should we ensure the survey_answer.answer matches one of them? I'd lean towards no (for this not to be the case, someone would have to manually POST an alternative value, as hopefully the front-end team will never do this) but wanted to flag it.

**Comment by mikelynch** [comment]
Dummy JSON for testing below:

## Case where no survey response
```json
{
    "survey_visit": {
        "user_id": 1,
        "completed": false,
        "home_id": 1,
        "survey_response_attributes": null
    }
}
```

## Case where have survey response
```json
{
    "survey_visit": {
        "user_id": 1,
        "completed": true,
        "home_id": 1,
        "survey_response_attributes": {
            "survey_id": 1,
            "survey_answers_attributes": {
                "1": {
                    "survey_question_id": 1,
                    "answer": "Yes"
                },
                "2": {
                    "survey_question_id": 2,
                    "answer": "No"
                }
            }
        }
    }
}
```

File: backend/app/models/survey_response.rb [hunk 1]
```diff
@@ -3,4 +3,6 @@
 class SurveyResponse < ApplicationRecord
   belongs_to :survey
   belongs_to :survey_visit
+  has_many :survey_answers, dependent: :destroy
```
[the remark this answers is not in the document]
Comment (mzagaja) [reply backend/app/models/survey_response.rb:6]: Yes, that is sensible.

**Comment by mzagaja** [comment]
> One remaining question: should we be trying to validate the survey_answers? I.e. if the corresponding survey_question is multiple choice, should we ensure the survey_answer.answer matches one of them? I'd lean towards no (for this not to be the case, someone would have to manually POST an alternative value, as hopefully the front-end team will never do this) but wanted to flag it.

I don't think we need to validate this. It's ok to leave as is.

> I'm also not sure whether we should be updating the data model for the survey_visit to reference surveyor_id rather than user_id. I think it probably should, but wanted to check.

Logically `surveyor_id` probably is a better fit. However, given a `surevyor` has to have a `user` and vice-versa it's not a material difference.

**Review by mzagaja — APPROVED** [review]
This is looking good. I would maybe consider adding the note in your comment as a request spec, if you want to make sure it's included in there. But we can do that after the fact pretty easily. The WiFi on my train is slow so I didn't manual test it locally, but if that cURL works, then I am satisfied with this update.

**Comment by mikelynch** [comment]
Thanks @mzagaja ! I'll make sure this is noted with the front-end team. I confirm those curl commands worked fine for me (i.e. got expected responses, and could see the data afterwards).

**Pull request merged**